src/track/annotation/AnnotationTrack.tsx

```diff
@@ -234,10 +238,41 @@ export class AnnotationTrack extends TrackObject<AnnotationTrackModel, Annotatio
                 annotation.gene.opacity = opacity;
 
                 annotation.name.visible = namesOpacity > 0;
-                annotation.name.opacity = namesOpacity;
+                // VALIS-8: no blending for gene name
+                annotation.name.opacity = Math.max(namesOpacity, 1.0);
                 annotation.name.x = 5
                 annotation.name.relativeX = Math.max(annotation.gene.relativeX, 0);
 
+                // VALIS-8: prevent collision between annotation names
+                const annotationNameWidth = annotation.name.getComputedWidth();
+                if (annotation.name.visible && annotationNameWidth) {
+                    for (let name of createdAnnotationNames) {
+                        if (annotation.name.y == name.y && gene.name != name.geneName) {
+                            // check gene's name to prevent multiple collision checking on the same gene
```

Comment:
[Let's talk about this]. The issue I have with this is that the text is misaligned from the genome maker. For example -  https://valis-8-fix-annot-text-leepc12.demo.encodedcc.org/experiments/ENCSR849LXI/

Make the valis smaller continuously and you will see the text move away from what it marks. Maybe hiding the text would be a better option.

Reply:
I think shifting gene labels gives enough information to users rather than just hiding overlapping gene labels. Gene labels are shifted but users know how to find the exact starting location of the gene by simply zooming out and navigating to left and right.